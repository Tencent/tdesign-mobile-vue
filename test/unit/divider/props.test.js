/* eslint-disable */

/**
 * 该文件为脚本自动生成文件，请勿随意修改。如需修改请联系 PMC
 * */

import { mount } from '@vue/test-utils';
import Divider from '../../../src/divider';
import { h } from 'vue';

describe('Divider', () => {
  describe(':props', () => {
    // 文本位置（仅在水平分割线有效）: left
    it('align', () => {
      const wrapper = mount(Divider, {
        propsData: {"align":"left"},
      });
      expect(wrapper.wrapperElement).toMatchSnapshot();
    });// 文本位置（仅在水平分割线有效）: right
    it('align', () => {
      const wrapper = mount(Divider, {
        propsData: {"align":"right"},
      });
      expect(wrapper.wrapperElement).toMatchSnapshot();
    });// 文本位置（仅在水平分割线有效）: center
    it('align', () => {
      const wrapper = mount(Divider, {
        propsData: {"align":"center"},
      });
      expect(wrapper.wrapperElement).toMatchSnapshot();
    });
    it('content', () => {
      const fn = jest.fn();
      const wrapper = mount(Divider, {
        propsData: {
          content: () => h('div', 'text')
        },
      });
      expect(wrapper.wrapperElement).toMatchSnapshot();
    });

    it('content', () => {
      const fn = jest.fn();
      const wrapper = mount(Divider, {
        propsData: {
          content: 'text'
        },
      });
      expect(wrapper.wrapperElement).toMatchSnapshot();
    });// 是否虚线（仅在水平分割线有效）: false
    it('dashed', () => {
      const wrapper = mount(Divider, {
        propsData: {"dashed":true},
      });
      expect(wrapper.wrapperElement).toMatchSnapshot();
    });
    it('default', () => {
      const fn = jest.fn();
      const wrapper = mount(Divider, {
        propsData: {
          default: () => h('div', 'text')
        },
      });
      expect(wrapper.wrapperElement).toMatchSnapshot();
    });

    it('default', () => {
      const fn = jest.fn();
      const wrapper = mount(Divider, {
        propsData: {
          default: 'text'
        },
      });
      expect(wrapper.wrapperElement).toMatchSnapshot();
    });// 分隔线类型有两种：水平和垂直: horizontal
    it('layout', () => {
      const wrapper = mount(Divider, {
        propsData: {"layout":"horizontal"},
      });
      expect(wrapper.wrapperElement).toMatchSnapshot();
    });// 分隔线类型有两种：水平和垂直: vertical
    it('layout', () => {
      const wrapper = mount(Divider, {
        propsData: {"layout":"vertical"},
      });
      expect(wrapper.wrapperElement).toMatchSnapshot();
    });// 分隔线颜色: 
    it('lineColor', () => {
      const wrapper = mount(Divider, {
        propsData: {"lineColor":""},
      });
      expect(wrapper.wrapperElement).toMatchSnapshot();
    });
  });
});