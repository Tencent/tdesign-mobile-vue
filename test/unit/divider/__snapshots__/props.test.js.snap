// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Divider :props align 1`] = `
<div
  class="t-divider t-divider--hairline"
  role="separator"
>
  <!---->
</div>
`;

exports[`Divider :props align 2`] = `
<div
  class="t-divider t-divider--hairline"
  role="separator"
>
  <!---->
</div>
`;

exports[`Divider :props align 3`] = `
<div
  class="t-divider t-divider--hairline"
  role="separator"
>
  <!---->
</div>
`;

exports[`Divider :props content 1`] = `
<div
  class="t-divider t-divider--hairline t-divider--content-center"
  role="separator"
>
  <div>
    text
  </div>
</div>
`;

exports[`Divider :props content 2`] = `
<div
  class="t-divider t-divider--hairline t-divider--content-center"
  role="separator"
>
  text
</div>
`;

exports[`Divider :props dashed 1`] = `
<div
  class="t-divider t-divider--hairline t-divider--dashed"
  role="separator"
>
  <!---->
</div>
`;

exports[`Divider :props default 1`] = `
<div
  class="t-divider t-divider--hairline t-divider--content-center"
  role="separator"
>
  <div>
    text
  </div>
</div>
`;

exports[`Divider :props default 2`] = `
<div
  class="t-divider t-divider--hairline t-divider--content-center"
  role="separator"
>
  text
</div>
`;

exports[`Divider :props layout 1`] = `
<div
  class="t-divider t-divider--hairline"
  role="separator"
>
  <!---->
</div>
`;

exports[`Divider :props layout 2`] = `
<div
  class="t-divider t-divider--hairline t-divider-vertical"
  role="separator"
>
  <!---->
</div>
`;

exports[`Divider :props lineColor 1`] = `
<div
  class="t-divider t-divider--hairline"
  role="separator"
>
  <!---->
</div>
`;
